forte/data/ontology/top.py

```diff
@@ -1219,15 +1235,36 @@ def payload_index(self) -> int:
     def uri(self) -> Optional[str]:
         return self._uri
 
-    def set_cache(self, data: Union[str, np.ndarray]):
+    @property
+    def cache_shape(self) -> Optional[Sequence[int]]:
+        return self._cache_shape
+
+    def set_cache(
+        self,
+        data: Union[str, np.ndarray],
+        cache_shape: Optional[Sequence[int]] = None,
+    ):
         """
-        Load cache data into the payload.
+        Load cache data into the payload and set the data shape.
 
         Args:
             data: data to be set in the payload. It can be str for text data or
                 numpy array for audio or image data.
+            cache_shape: the shape of the data. Its representation varies based
+                on the modality and its length.
+                For text data, it is the length of the text.[length, text_embedding_dim]
+                For audio data, it is the length of the audio. [length, audio_embedding_dim]
+                For image data, it is the shape of the image. [height, width,
+                channel]
+                For example, for image data, if the cache_shape length is 2, it
+                means the image data is a 2D image [height, width].
         """
         self._cache = data
+        if isinstance(data, np.ndarray):
```

Comment:
Looks like we can also set `cache_shape` when `data` is a string.

Reply:
fixed it.